source/examples/rapids-sagemaker-hpo/notebook.ipynb

```diff
@@ -1,7 +1,6 @@
 {
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #1.    image_base = "sagemaker-rapids-mnmg-100"
</pre><p> </p><p>The default role that AWS attaches when you launch a SageMaker Notebook Instance has the following permissions included:</p><p> </p><p> </p>

<pre class="ql-syntax" spellcheck="false">{
    "Effect": "Allow",
    "Action": [
        "ecr:BatchDeleteImage",
        "ecr:CompleteLayerUpload",
        "ecr:CreateRepository",
        "ecr:DeleteRepository",
        "ecr:InitiateLayerUpload",
        "ecr:PutImage",
        "ecr:UploadLayerPart"
    ],
    "Resource": [
        "arn:aws:ecr:*:*:repository/sagemaker-*"
    ]
}
</pre><p> </p><p> </p><p>So changing the image name to begin with <code>sagemaker-*</code> allows it to be pushed to ECR without needing to modify any permissions.</p><p> </p><p>Without this change, running this under a SageMaker Notebook Instance with the default permissions fails at the <code>docker push</code> step, with an error like this:</p><p> </p>

<pre class="ql-syntax" spellcheck="false">64186e1: Waiting g denied:
User: arn:aws:sts::561241433344:assumed-role/AmazonSageMakerServiceCatalogProductsUseRole/SageMaker
is not authorized to perform: ecr:InitiateLayerUpload
on resource: arn:aws:ecr:us-west-2:561241433344:repository/rapids-sagemaker-mnmg-100
because no identity-based policy allows the ecr:InitiateLayerUpload action
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/deployment/pull/381/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='65'/>

Reply:
<p>Oh nice!</p>